src/testWithSpringBoot_2_7/java/org/openrewrite/java/spring/boot2/RemoveSolrAutoConfigurationExcludeTest.java

```diff
@@ -0,0 +1,194 @@
+              @SpringBootApplication(exclude = {SecurityAutoConfiguration.class})
+              public class Application {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeEntireArray() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import org.springframework.boot.autoconfigure.SpringBootApplication;
+              import org.springframework.boot.autoconfigure.solr.SolrAutoConfiguration;
+
+              @SpringBootApplication(exclude = { SolrAutoConfiguration.class })
+              class Application {
+              }
+              """,
+            """
+              import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+              @SpringBootApplication
+              class Application {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeArgument() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import org.springframework.boot.autoconfigure.SpringBootApplication;
+              import org.springframework.boot.autoconfigure.solr.SolrAutoConfiguration;
+
+              @SpringBootApplication(exclude = SolrAutoConfiguration.class)
+              class Application {
+              }
+              """,
+            """
+              import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+              @SpringBootApplication
+              class Application {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeFullyQualifiedArgument() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+              @SpringBootApplication(exclude = org.springframework.boot.autoconfigure.solr.SolrAutoConfiguration.class)
+              class Application {
+              }
+              """,
+            """
+              import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+              @SpringBootApplication
+              class Application {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeFromEnableAutoConfiguration() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+              import org.springframework.boot.autoconfigure.solr.SolrAutoConfiguration;
+
+              @EnableAutoConfiguration(exclude = SolrAutoConfiguration.class)
+              class Application {
+              }
+              """,
+            """
+              import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+
+              @EnableAutoConfiguration
+              class Application {
+              }
+              """
+          )
+        );
+    }
+
+    @Nested
+    class NoChange {
```

Comment:
When the number of test cases grows, I like to group them logically for the ones that ought not make any changes. That quickly shows the invariants and makes it logical what to add where.